rest/route/sns.go

```diff
@@ -42,94 +44,104 @@ type awsSns struct {
 	payload     sns.Payload
 }
 
-func makeAwsSnsRoute(sc data.Connector, env evergreen.Environment, queue amboy.Queue) gimlet.RouteHandler {
-	return &awsSns{
+func makeBaseSNS(sc data.Connector, env evergreen.Environment, queue amboy.Queue) baseSNS {
+	return baseSNS{
 		sc:    sc,
 		env:   env,
 		queue: queue,
 	}
 }
 
-func (aws *awsSns) Factory() gimlet.RouteHandler {
-	return &awsSns{
-		sc:    aws.sc,
-		env:   aws.env,
-		queue: aws.queue,
-	}
-}
-
-func (aws *awsSns) Parse(ctx context.Context, r *http.Request) error {
-	aws.messageType = r.Header.Get("x-amz-sns-message-type")
+func (sns *baseSNS) Parse(ctx context.Context, r *http.Request) error {
+	sns.messageType = r.Header.Get("x-amz-sns-message-type")
 
 	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
 		return errors.Wrap(err, "problem reading body")
 	}
-	if err = json.Unmarshal(body, &aws.payload); err != nil {
+	if err = json.Unmarshal(body, &sns.payload); err != nil {
 		return errors.Wrap(err, "problem unmarshalling payload")
 	}
 
-	if err = aws.payload.VerifyPayload(); err != nil {
+	if err = sns.payload.VerifyPayload(); err != nil {
 		msg := "AWS SNS message failed validation"
 		grip.Error(message.WrapError(err, message.Fields{
 			"message": msg,
-			"payload": aws.payload,
+			"payload": sns.payload,
 		}))
 		return errors.Wrap(err, msg)
 	}
 
 	return nil
 }
 
-func (aws *awsSns) Run(ctx context.Context) gimlet.Responder {
-	// Subscription/Unsubscription is a rare action that we will handle manually and will be logged to splunk given the logging level.
-	switch aws.messageType {
+type ec2SNS struct {
+	baseSNS
```

Comment:
Doesn't need to be a pointer.

Reply:
But doesn't `Parse` mutate the struct? If it is a value receiver, the payload won't get stored as we expect. Or does it work because `ec2SNS` is a pointer?

Reply:
Actually ignore this, I just realized that what matter is that `Parse` takes a pointer receiver. Sorry, I confused myself.